pkgs/by-name/br/breakpointHook/package.nix

```
@@ -1,6 +1,37 @@
-{ stdenv, makeSetupHook }:
+{
+  lib,
+  stdenv,
+
+  bash,
+  bashInteractive,
+  coreutils,
+  makeSetupHook,
+  procps,
+  util-linux,
+  writeShellScriptBin,
+}:
+
+let
+  attach = writeShellScriptBin "attach" ''
+    export PATH="${
+      lib.makeBinPath [
+        bash
+        coreutils
+        procps # needed for pgrep
+        util-linux # needed for nsenter
+      ]
+    }"
+    exec bash ${./attach.sh}
+  '';
+in
 
 makeSetupHook {
   name = "breakpoint-hook";
   meta.broken = !stdenv.buildPlatform.isLinux;
+  substitutions = {
+    attach = "${attach}/bin/attach";
+    # The default interactive shell in case $debugShell is not set in the derivation.
+    # Can be overridden to zsh or fish, etc.
+    debugShell = "${bashInteractive}/bin/bash";
```

I would make this slightly longer to avoid potential namespace collision:

```suggestion
    breakpointHookShell = "${bashInteractive}/bin/bash";
```

---

So much typing ;) I want to be quick when debugging. There are currently zero uses of `debugShell` in nixpkgs.

---

How about `failureShell`?

---

Does a user has to ever type this themselves?

---

Yes, whenever you want to override the interactive shell used by breakpointHook you would set this variable in your derivation.

---

It was confusing that this variable has the same name as the build time `debugShell` which is read by the hook. I changed it now.